zerver/lib/mention.py

```diff
@@ -271,7 +271,12 @@ def init_user_group_data(self, realm_id: int, content: str) -> None:
         if user_group_names:
             for group in UserGroup.objects.filter(
                 realm_id=realm_id, name__in=user_group_names, is_system_group=False
-            ).prefetch_related("direct_members"):
+            ).prefetch_related(
+                Prefetch(
+                    "direct_members",
+                    queryset=UserProfile.objects.filter(realm_id=realm_id, is_active=True),
+                )
+            ):
```

Comment:
Did this instead of using `group.direct_members.filter(is_active=True)` as that would defeat the purpose of using `prefetch_related`. This makes sure that the number of queries remain the same as it was before.

Reply:
OK, please record the reason we're doing this in a comment...

Hmm, also, maybe we can tweak this to only prefetch the `ID` field, not the whole object and use a more efficient query? It looks like a `group.direct_members.all().values_list("id", flat=True)` might help? Would want to print-debug the SQL used to be sure, but I think this could be a significant little performance tweak.

Reply:
We cannot use `values_list` or `values` in the `queryset` arg to `Prefetch` and if we do `group.direct_members.all().values_list("id", flat=True)` then it generates an extra query.

Reply:
OK, probably a comment is the right answer here.